mm2src/mm2_main/src/lp_ordermatch.rs

```diff
@@ -2977,79 +3016,174 @@ fn lp_connect_start_bob(ctx: MmArc, maker_match: MakerMatch, maker_order: MakerO
             },
         };
 
-        if ctx.use_trading_proto_v2() {
-            let secret_hash_algo = detect_secret_hash_algo(&maker_coin, &taker_coin);
-            match (maker_coin, taker_coin) {
+        let alice_swap_v = maker_match.request.swap_version;
+        // if taker order doesn't have swap version or taker uses legacy swap protocol, start legacy swap
+        if alice_swap_v.is_none() || alice_swap_v == Some(1) {
+            let params = LegacySwapParams {
+                maker_coin: &maker_coin,
+                taker_coin: &taker_coin,
+                uuid: &uuid,
+                my_conf_settings: &my_conf_settings,
+                my_persistent_pub: &my_persistent_pub,
+                maker_amount: &maker_amount,
+                taker_amount: &taker_amount,
+                locktime: &lock_time,
+            };
+            start_maker_legacy_swap(&ctx, maker_order, alice, secret, params).await;
```

Comment:
we start legacy swap in this case and also in the fall back else case of the condition below for v2 swaps, we could merge these

Reply:
Done https://github.com/KomodoPlatform/komodo-defi-framework/pull/2261/commits/f817f5cd97178af169afb14cfd8d559ed84c6700